src/components/Login.js

```diff
@@ -1,32 +1,94 @@
 import { useState } from 'react';
+import { useTheme } from '../App';
+import CheckBox from 'react-animated-checkbox';
 import * as Api from '../Api';
 import '../styles/Login.css';
 
-const Login = () => {
-    const [usernameInput, setUsernameInput] = useState('');
+const Login = ({ setToken, darkMode }) => {
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [message, setMessage] = useState('');
     const [passwordInput, setPasswordInput] = useState('');
-    const [messageText, setMessageText] = useState('');
+    const [showPassword, setShowPassword] = useState(false);
 
     const login = async () => {
-        const result = await Api.login(usernameInput, passwordInput)
+        console.log({ usernameInput: username, passwordInput: password });
+        if (!username && password) {
+            setMessage("Username can't be empty!");
+            return;
+        }
+        else if (username && !password) {
+            setMessage("Password can't be empty!");
+            return;
+        }
+        else if (!username && !password) {
+            setMessage("Password and username can't be empty!");
+            return;
+        }
 
-        if (result) {
-            setMessageText('Wrong username or password.');
+        try {
+            const result = await Api.login(username, password);
```

Comment:
You can add better errors here (so the user knows exactly what's wrong) if you change `Api.login` according to my suggestion

Reply:
What do you recommend?

Reply:
Oh sorry, forgot about that!
```js
const login = async (username, password) => {
    /* ... */
    if (success) {
        return token;
    }
    else {
        return errorCode;
    }
}
```

With this you can check against status codes (there are only a few that can be returned from a login failiure) and provide better errors.

Reply:
Can you tell me all return codes?